### PR TITLE
Single Line and Multi Line Input Text

### PR DESCRIPTION
**Pull request by praspra2**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented the Single Line and Multi Line Input Text for the Input Text element using the model based approach.
- https://app.vidcast.io/share/d51d251d-be83-4374-9b9a-aa0ebf44bd03

image
image

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

**Comment by soumykun** [comment]
Review in progress.

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.h [hunk 1]
```diff
@@ -54,6 +64,12 @@ namespace AdaptiveCardQmlEngine
...
         std::shared_ptr<AdaptiveCardConfig> mCardConfig;
         AdaptiveCardEnums::AdaptiveCardTheme mAdaptiveCardTheme;
-        std::string m_lang; 
+        std::string m_lang;
+        int m_HeightEstimate{0};
+        std::vector<AdaptiveWarning> m_warnings;
+        std::string m_defaultId;
+        int m_DefaultIdCounter{0};
+        std::vector<std::string> m_RequiredInputElementsIdList;
+        std::map<std::string, std::string> m_inputElementList;
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.h:73]: Lets not mix up the variable naming conventions...lets follow one style.

File: source/qml_v2/AdaptiveCardQmlEngine/models/TextInputModel.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    if (input->GetHeight() == AdaptiveCards::HeightType::Stretch)
+    {
+        mHeightStreched = input->GetHeight() == AdaptiveCards::HeightType::Stretch ? "true" : "false";
+    }
+    if (input->GetIsMultiline())
+    {
+        mIsMultiLineText = true;
+        // addActions();
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/TextInputModel.cpp:53]: commented code ?

File: source/qml_v2/AdaptiveCardQmlEngine/models/TextInputModel.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    else
+    {
+        mIsMultiLineText = false;
+        // addActions();
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/TextInputModel.cpp:58]: Commented code ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/CardConstants.qml [hunk 4]
```diff
@@ -1,6 +1,6 @@
-import "AdaptiveCardUtils.js" as AdaptiveCardUtils
+import "JSUtils/AdaptiveCardUtils.js" as AdaptiveCardUtils
 import QtQuick 2.15
-pragma Singleton
+//pragma Singleton
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/CardConstants.qml:4]: Lets connect on this !

File: source/qml_v2/AdaptiveCardQmlEngine/utils/Utils.h [hunk 5]
```diff
@@ -3,52 +3,59 @@
...
+		static std::string& ToLower(std::string& str);
+		static bool hasNonAlphaNumeric(const std::string& id);
+		static const std::string HandleKeywords(const std::string& id);
+
+		template <class T, class U>
+		static bool IsInstanceOfSmart(U u);
+
+		template <class T, class U>
+		static bool IsInstanceOf(U u);
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/utils/Utils.h:33]: Indentation looks off !

**Review by soumykun — APPROVED** [review]

**Review by ramkrishnajana — APPROVED** [review]

**Pull request merged**